util/window/policy.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package window
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/exp/constraints"
+)
+
+// RollingPolicy is a policy for ring windows based on time duration.
+// RollingPolicy moves bucket offset with time duration.
+// e.g. If the last point is appended one bucket duration ago,
+// RollingPolicy will increment current offset.
+type RollingPolicy[T constraints.Integer | constraints.Float] struct {
+	mu     sync.RWMutex
+	size   int
+	window *Window[T]
+	offset int
```

Comment:
the offset points to the oldest bucket or newest bucket? (I guess it's newest) please add comment